service/share/full_availability.go

```diff
@@ -6,27 +6,49 @@ import (
 
 	"github.com/ipfs/go-blockservice"
 	format "github.com/ipfs/go-ipld-format"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p/p2p/discovery/routing"
 
 	"github.com/celestiaorg/celestia-node/ipld"
 )
 
-// fullAvailability implements Availability using the full data square
+// FullAvailability implements Availability using the full data square
 // recovery technique. It is considered "full" because it is required
 // to download enough shares to fully reconstruct the data square.
-type fullAvailability struct {
-	rtrv *ipld.Retriever
+type FullAvailability struct {
```

Comment:
To be able to call Start/Stop without extending `Availability` interface. Returning unexported struct will cause linter warning.

Reply:
I'd rather add `Start/Stop` to the availability interface.

Then we can unexport CacheAvailability and call `Close` -> `Stop`

Reply:
Lets' not add Start/Stop to the interfaces.

Reply:
Or generally, let's not add to interfaces things that can be omitted and implementation-specific.

Reply:
Agree with @Wondertan.

Reply:
I think lifecycle is one of those cases that is applicable to almost everything, so I'm fine having Availability implement Start/Stop. Yes generally I also oppose implementations having empty methods, but the only empty method for Start/Stop now would be CacheAvailability Start which is ok IMO bc it would just call underlying avail `Start` so that we only need to add the cache availability lifecycle to fx.

CacheAvailability `Close` -> `Stop` and can wrap underlying avail stop

Reply:
There are a few other points here that we should also consider. Let's extract this discussion to some issue and not block the PR on it.